content/en/blog/2020/on-demand-egress-gateways/index.md

```diff
@@ -0,0 +1,500 @@
+target_release: 1.8.0
+---
+
+At [`Deutsche Telekom::PAN-NET`](https://pan-net.cloud/aboutus), we have embraced `Istio` as the umbrella to cover our services.
+
+Unfortunately, there are services which have not been migrated to `kubernetes` yet or, simply, cannot be. Therefore, those ones have to remain as external services.
+
+## Scenario
+
+If you are familiar with Istio, one of the offered ways to connect to external services is through [Egress Gateway](/docs/tasks/traffic-management/egress/egress-gateway/).
+
+However, if you want to satisfy the [Single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), you will need to deploy multiple and individual (1..N) `Egress Gateways` as the picture shows.
+
+{{< image width="75%" ratio="45.34%"
+    link="./on-demand-egress-gateway-overview.svg"
+    alt="Overview multiple Egress Gateways"
+    caption="Overview multiple Egress Gateways"
+    >}}
+
+With this model, one `Egress Gateway` is in charge of only one upstream service.
+
+Although the `IstioOperator` spec allows you to deploy multiple `Egress Gateways`, the manifest can become unmanageable.
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+[...]
+spec:
+    egressGateways:
+    - name: egressgateway-1
+      enabled: true
+    - name: egressgateway-2
+      enabled: true
+    [egressgateway-3, egressgateway-4 ,...]
+    - name: egressgateway-N
+      enabled: true
+[...]
+{{< /text >}}
+
+As a benefit of decoupling Egress Getaways from IstioOperator manifest, you have enabled the possibility of setting up **custom readiness probes** to have both services (gateway and upstream service) aligned.
+
+As well, you can apply the sidecar pattern to inject, for example, OPA into the pod to perform authorization with complex rules ([OPA envoy plugin](https://github.com/open-policy-agent/opa-envoy-plugin)).
+
+{{< image width="75%" ratio="45.34%"
+    link="./on-demand-egress-gateway-authz.svg"
+    alt="Authorization with OPA and `healthcheck` to external service"
+    caption="Authorization with OPA and `healthcheck` to external"
+    >}}
+
+At this point, you might be convinced that this is the right path to go.
+
+## Solution
+
+There are several ways to achieve this task:
+
+- Create multiple `IstioOperator` (simple solution)
+- Create an operator (recommended for a second iteration in the development process)
+
+At **PAN-NET**, we have created our own operator with the [Operator SDK - Ansible type](https://sdk.operatorframework.io/docs/building-operators/).
+
+However, you want to start simple. Thus, better you start with the `IstioOperator` option.
+
+{{< quote >}}
+Yes! `Istio 1.8.0` introduced the possibility to have fine-grained control over the objects that `IstioOperator` deploys. This gives you the opportunity to patch them as you wish. Exactly what you need to deploy on demand `Egress Gateways`.
+{{< /quote >}}
```

Comment:
Does this invalidate everything you say below?

(We now seem to have three methods to solve this: the one you propose, the new 1.8 method, and the mysterious Operator method you actually use!)

Reply:
As commented above, with the new proposal, nothing else is mentioned so that reader is focused only in the proposed solution.

The reason of this sentence is that: yes! The version of the post was for 1.7. And now, with 1.8, it was very grateful to find out that you can deploy multiple Operators (although you have to generate the manifests upfront)